src/features/authentication/authSlice.js

```diff
@@ -0,0 +1,268 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import axios from "axios";
+
+export const refreshFromUsername = createAsyncThunk(
+  "auth/refreshFromUsername",
+  async (username, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_BACKEND}/users/username`,
+        {
+          username,
+        }
+      );
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+export const loginUser = createAsyncThunk(
+  "auth/loginUser",
+  async ({ email, password }, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_BACKEND}/users/login`,
+        {
+          email,
+          password,
+        }
+      );
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+export const signupUser = createAsyncThunk(
+  "auth/signupUser",
+  async ({ name, email, password }, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_BACKEND}/users/signup`,
+        {
+          name,
+          email,
+          password,
+        }
+      );
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+export const updateUserDetails = createAsyncThunk(
+  "auth/updateUserDetails",
+  async ({ userUpdates, userId }, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_BACKEND}/users/${userId}`,
+        {
+          userUpdates,
+        }
+      );
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+export const followUser = createAsyncThunk(
+  "auth/followUser",
+  async (userId, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_BACKEND}/users/follow`,
+        {
+          userId,
+        }
+      );
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+export const unfollowUser = createAsyncThunk(
+  "auth/unfollowUser",
+  async (userId, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_BACKEND}/users/unfollow`,
+        {
+          userId,
+        }
+      );
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+let login = JSON.parse(localStorage.getItem("ggLogin"));
+let local_token = login?.token || null;
+let local_userData = login?.userData || null;
+let userLoginStatus = local_token ? true : false;

```

Comment:
can get rid of these variables, and directly assign to initial state using only `login` variable